pkl-core/src/main/java/org/pkl/core/ast/type/TypeNode.java

```diff
@@ -1342,6 +1373,14 @@ public Object executeEagerly(VirtualFrame frame, Object value) {
       throw typeMismatch(value, BaseModule.getMapClass());
     }
 
+    @Override
+    public Object executeInDefinitionContext(VirtualFrame frame, Object value) {
+      if (value instanceof VmMap vmMap) {
+        return evalInDefinitionContext(frame, vmMap);
+      }
+      throw typeMismatch(value, BaseModule.getMapClass());
```

Comment:
For consistency and to save on single use (albeit `private`) helpers; why not follow the same pattern as in `VmList`?

Reply:
I did this for consistency with `execute` and `executeEager`. I don't think it makes sense to only inline it here. Either we remove `eval` and `evalEager` or we leave all three.